controllers/clusterresources/helpers.go

```diff
@@ -119,3 +123,76 @@ func subnetsEqual(subnets1, subnets2 []string) bool {
 
 	return true
 }
+
+type ClusterIdProvider interface {
+	client.Object
+	GetClusterID() string
+	GetDatacentreID(cdcName string) string
+}
+
+func GetDatacentreID(cl client.Client, ctx context.Context, ref *v1beta1.ClusterRef) (string, error) {
+	var obj ClusterIdProvider
+	ns := types.NamespacedName{
+		Namespace: ref.Namespace,
+		Name:      ref.Name,
+	}
+
+	switch ref.ClusterKind {
+	case models.RedisClusterKind:
+		obj = &clustersv1beta1.Redis{}
+	case models.OpenSearchKind:
+		obj = &clustersv1beta1.OpenSearch{}
+	case models.KafkaKind:
+		obj = &clustersv1beta1.Kafka{}
+	case models.CassandraKind:
+		obj = &clustersv1beta1.Cassandra{}
+	case models.PgClusterKind:
+		obj = &clustersv1beta1.PostgreSQL{}
+	case models.ZookeeperClusterKind:
+		obj = &clustersv1beta1.Zookeeper{}
+	case models.CadenceClusterKind:
+		obj = &clustersv1beta1.Cadence{}
+	case models.KafkaConnectClusterKind:
+		obj = &clustersv1beta1.KafkaConnect{}
+	}
+	err := cl.Get(ctx, ns, obj)
+	if err != nil {
+		return "", err
+	}
+
+	return obj.GetDatacentreID(ref.CDCName), nil
+}
+
+func GetClusterID(cl client.Client, ctx context.Context, ref *v1beta1.ClusterRef) (string, error) {
+	var obj ClusterIdProvider
+	ns := types.NamespacedName{
+		Namespace: ref.Namespace,
+		Name:      ref.Name,
+	}
+
+	switch ref.ClusterKind {
+	case models.RedisClusterKind:
+		obj = &clustersv1beta1.Redis{}
+	case models.OpenSearchKind:
+		obj = &clustersv1beta1.OpenSearch{}
+	case models.KafkaKind:
+		obj = &clustersv1beta1.Kafka{}
+	case models.CassandraKind:
+		obj = &clustersv1beta1.Cassandra{}
+	case models.PgClusterKind:
+		obj = &clustersv1beta1.PostgreSQL{}
+	case models.ZookeeperClusterKind:
+		obj = &clustersv1beta1.Zookeeper{}
+	case models.CadenceClusterKind:
+		obj = &clustersv1beta1.Cadence{}
+	case models.KafkaConnectClusterKind:
+		obj = &clustersv1beta1.KafkaConnect{}
+	}
```

Comment:
Let's add the default case if the customer sets an invalid kind

Reply:
agree, refactored